### ISSUE TITLE
Problem with Postpone

### ISSUE THREAD

**Issue by jjwhite** [recurly/recurly-client-net#214]
I'm seeing an issue with the postpone method where if I send it the current datetime or any datetime that is earlier than 10 hours from now I receive the error message: "The information being saved is not valid"  

Here is a simple example:
```
 DateTime utc = DateTime.UtcNow.AddMinutes(1);
 subscription.Postpone(utc);
```

If I change it to be 10 hours in the future it works fine (but anything less than 10 hours also fails):
```
 DateTime utc = DateTime.UtcNow.AddHours(10);
 subscription.Postpone(utc);
```
The current state of the subscription in recurly:
StartDate: Jan 24, 2017 9:53 AM EST
Next Invoice: Feb 28, 2017 7:00 AM EST

I tried capturing the actual request being made to recurly but it wasn't showing up in Fiddler even after adding a defaultproxy to my web.config.

**Comment by 1TheMuffinMan** [comment]
@jjwhite Thanks for opening this issue separately.

What I found when testing this is that the date for `Postpone` must be a certain amount of time after the `activation/current period started` date. 

For example:

- If the the postpone date is ~ <10hours from the activation or current period started date it will not work
- However, if you did `subscription.CurrentPeriodEndsAt.Value.AddMinutes(1)` that would work if the current datetime was 10 hours away from that date.

Is there a specific use case whereby you need to create a subscription and then change the postpone date so that the subscription is only active for less than 10 hours total?

**Comment by jjwhite** [comment]
Thanks @1TheMuffinMan.  No, I don't have a specific use case that would require a subscription that is active for less than 10 hours so I think I can fix this by ensuring that the postpone date is at least 10 hours after the started date.

**Comment by jjwhite** [comment]
We just did a test with a Recurly user whose startdate was Nov. 11, 2016 and it still failed until I added 10 hours to the current datetime.

**Comment by 1TheMuffinMan** [comment]
I think if this is a behavior that you'd like changed in the API itself it would need to be brought up with Recurly support as there's no way to get around that using the .NET library. Please let me know if there is anything else I can do to assist.

At the very least, I can see about getting those constraints added to the documentation so it's more clear.

**Comment by bhelx** [comment]
@1TheMuffinMan I can check the logic on the server to give a better description of how this validation works.

@jjwhite Could you explain what you are trying to accomplish from your customer's perspective? I think we may be able to help you with a cleaner solution.

**Comment by bhelx** [comment]
@1TheMuffinMan @jjwhite When I run the same code:

```csharp
            try {
                var sub = Subscriptions.Get("3b1c9a5c19f46a8e205abe409ebe3c01");
                DateTime utc = DateTime.UtcNow.AddMinutes(100);
                sub.Postpone(utc);
            } catch (ValidationException ex)
            {
                foreach(Error err in ex.Errors)
                {
                    Console.WriteLine(err.ToString());
                }
            }
```

I'm getting the error symbol: `past_date` desc: `must be in the future`. I believe this is because it's expecting the next_renewal_date to be at least `tomorrow`. So you may have found that 10 hours worked for you because that was the next day in UTC. In other words, it has to be a future date, not a future datetime. 

I believe we have a granularity of days here because we don't guarantee that a subscription is renewed at a certain time, only on a certain day. So if you want to postpone the minimum amount of time, add one day instead of 10 hours:

```csharp
DateTime utc = DateTime.UtcNow.AddDays(1);
```

**Comment by bhelx** [comment]
I think I may try to make these error messages more descriptive as well.

**Comment by 1TheMuffinMan** [comment]
@bhelx Thanks for looking into it. In my tests I observed that the postpone day just needed to be a certain amount of time away from the current date `DateTime.Utc.Now`. You could actually set the postpone date before the  `current_period_ends_at` as long as it's still a number of hours away of what the current date/time is (at least that is what I remember). 

The only use case I can think of is a customer calls in and asks to have their renewal date changed to today.

**Comment by 1TheMuffinMan** [comment]
@jjwhite What we're planning to do is add a check to methods that take DateTime as an argument to make sure the `DateTimeKind == UTC`. This will it will be more foolproof. It will go into the release 1.5 branch that we are using for breaking changes (currently async/wait feature is in there). I'll keep this issue open until we merge it into master down the road.

**Comment by jjwhite** [comment]
Sounds good @1TheMuffinMan

**Comment by aaron-suarez** [comment]
I tried to reproduce this issue on the current version of the library and API, and it does not seem to be an issue anymore. It's very possible that it was fixed when we made the move from `next_renewal_date` to `next_bill_date`. I did, however, notice that if you pass in a DateTime with `DateTimeKind == Local`, the API still treats that as a UTC date. This problem was solved for `Usage`'s DateTime in #345, so I'm going to make that same change here. I hope that change will prevent any future problems we may have with the Postpone method.

@jjwhite If you're still seeing this problem on current versions of the API and client library, let me know and I will look a bit further into it.
